lib/rules/no-loop-func.js

```diff
@@ -185,6 +199,10 @@ module.exports = {
                 return;
             }
 
+            if (!node.async && !node.generator && isIIFE(node.parent)) {
```

Comment:
This would be a false negative:

```js
/* eslint no-loop-func: "error" */

var arr = [];

for (var i = 0; i < 5; i++) {
    arr.push((f => f)(() => i)); // false negative, this is not a call of `() => i`
}

console.log(arr.map(f => f())); // [ 5, 5, 5, 5, 5 ]

```

Parent of `() => i` is a function call, but not a call of that function (`() => i` is an argument, not callee).